src/test/groovy/io/github/cosmicsilence/scalafix/ScalafixPluginFunctionalTest.groovy

```diff
@@ -151,6 +152,80 @@ sourceSets {
         buildResult.output.contains(':checkScalafixIntegTest SKIPPED')
     }
 
+    def 'scalafix semantic rule RemoveUnused fixes scala code'() {
+        given:
+        scalafixConfFile.write "rules = [ RemoveUnused ]"
+
+        when:
+        runGradleTask('scalafix', [ ])
+
+        then:
+        scalaSrcFile.getText() == ScalafixTestConstants.FIXED_SCALA_CODE
+    }
+
+    def 'checkScalafix semantic rule RemoveUnused fails a build'() {
+        given:
+        scalafixConfFile.write "rules = [ RemoveUnused ]"
+
+        when:
+        runGradleTask('checkScalafix', [ ])
+
+        then:
+        // code is not changed on disk
+        scalaSrcFile.getText() == ScalafixTestConstants.FLAWED_SCALA_CODE
+        final UnexpectedBuildFailure err = thrown()
+        err.message.contains("A file on disk does not match the file contents if it was fixed with Scalafix")
+    }
+
+
+    def 'scalafix semantic rule RemoveUnused skips excluded scala files'() {
```

Comment:
sug.: 'scalafix should skip excluded source files'

Maybe we should have the same test for `checkScalafix`